taxi/views.py

```diff
@@ -79,6 +89,24 @@ class CarDeleteView(LoginRequiredMixin, generic.DeleteView):
     success_url = reverse_lazy("taxi:car-list")
 
 
+def car_assign_driver(request: HttpRequest, pk: int) -> HttpResponse:
+    car = Car.objects.get(id=pk)
+    car.drivers.add(request.user)
+    context = {
+        "car": car
+    }
+    return render(request, "taxi/car_detail.html", context)
+
+
+def car_delete_driver(request: HttpRequest, pk: int) -> HttpResponse:
+    car = Car.objects.get(id=pk)
```

Comment:
Replaces Car.objects.get(id=pk) with get_object_or_404(Car, id=pk). This ensures that if the car with the given ID does not exist, a 404 error page is shown instead of raising an unhandled exception.